DAQ_System/Source/Adapter/DataLogger/SdDataLogger.hpp

```diff
@@ -0,0 +1,48 @@
+/*
+* SD Data Logger
+* Authors:        Cristian Cruz
+*                 Ethan Fischer
+*
+* Email:          cris14.alex@gmail.com
+*                 ejf180001@utdallas.edu
+* 
+* (c) 2022 Dallas Formula Racing - Embedded Firmware Team
+* Formula SAE International Collegiate Chapter
+* GPL-3.0 License
+*/
+
+#ifndef DATA_LOGGER_H
+#define DATA_LOGGER_H
+
+#include "mbed.h"
+#include "FATFileSystem.h"
+#include "SDBlockDevice.h"
+
+#include "Application/I_Data_Logger.hpp"
+
+namespace adapter {
+
+class SdDataLogger : public application::I_Data_Logger {
+    public:
+        SdDataLogger(PinName mosi, PinName miso, PinName sck, PinName cs);
+        virtual ~SdDataLogger();
+        
+        virtual uint8_t Mount(FileSystem*, BlockDevice*) override;
+        virtual uint8_t Unmount(FileSystem*) override;
+        virtual uint8_t FileOpen(FILE**, char*) override;
+        virtual uint8_t FileClose(FILE*) override;
+        virtual uint8_t FileWrite(FILE*, const char*) override;
+
+        char write_buffer; // write buffer
```

Comment:
- Initialize `write_buffer` with `kBlockSectorByteSize`
- Initialize `kBlockSectorByteSize` to 512 here instead
- remove all other instances of `kBlockSectorByteSize` in the source file

Reply:
Done